alicloud/resource_alicloud_route_table_association_test.go

```diff
@@ -0,0 +1,174 @@
+package alicloud
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/vpc"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("alicloud_route_table_association", &resource.Sweeper{
+		Name: "alicloud_route_table_association",
+		F:    testSweepRouteTableAssociation,
+		// When implemented, these should be removed firstly
+		Dependencies: []string{},
+	})
+}
+
+func testSweepRouteTableAssociation(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting Alicloud client: %s", err)
+	}
+	conn := client.(*AliyunClient)
+
+	prefixes := []string{
+		"tf-testAcc",
+		"tf_testAcc",
+		"tf_test_",
+		"tf-test-",
+		"testAcc",
+	}
+
+	var routeTables []vpc.RouterTableListType
+	req := vpc.CreateDescribeRouteTableListRequest()
+	req.RegionId = conn.RegionId
+	req.PageSize = requests.NewInteger(PageSizeLarge)
+	req.PageNumber = requests.NewInteger(1)
+	for {
+		resp, err := conn.vpcconn.DescribeRouteTableList(req)
+		if err != nil {
+			return fmt.Errorf("Error retrieving RouteTables: %s", err)
+		}
+		if resp == nil || len(resp.RouterTableList.RouterTableListType) < 1 {
+			break
+		}
+		routeTables = append(routeTables, resp.RouterTableList.RouterTableListType...)
+
+		if len(resp.RouterTableList.RouterTableListType) < PageSizeLarge {
+			break
+		}
+
+		if page, err := getNextpageNumber(req.PageNumber); err != nil {
+			return err
+		} else {
+			req.PageNumber = page
+		}
+	}
+
+	for _, vtb := range routeTables {
+		name := vtb.RouteTableName
+		id := vtb.RouteTableId
+		skip := true
+		for _, prefix := range prefixes {
+			if strings.HasPrefix(strings.ToLower(name), strings.ToLower(prefix)) {
+				skip = false
+				break
+			}
+		}
+		if skip {
+			log.Printf("[INFO] Skipping Route Table: %s (%s)", name, id)
+			continue
+		}
+		log.Printf("[INFO] Unassociating Route Table: %s (%s)", name, id)
+		req := vpc.CreateUnassociateRouteTableRequest()
+		req.RouteTableId = id
+		if _, err := conn.vpcconn.UnassociateRouteTable(req); err != nil {
+			log.Printf("[ERROR] Failed to unassociate Route Table (%s (%s)): %s", name, id, err)
+		}
+	}
```

Comment:
I think there should wait several seconds to ensure `UnassociateRouteTable` is success.